metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java

```diff
@@ -299,7 +299,11 @@ public ResponseEntity<RegisterSchemaResponse> register(
             })
         .orElseGet(
             () -> {
-              log.error("Couldn't find topic with name {}.", topicName);
+              if (topicName.matches("^[a-zA-Z0-9._-]+$")) {
```

Comment:
Could you explain what is the security issue with printing the topic name?

Reply:
Not exactly a security issue, but codeql was giving `log injection` vulnerability.
Similiar with other log changes as well.

Reply:
topicName needs to be sanitized before logging.  If it cannot be sanitized, we are not logging it.

Reply:
why does it need to be sanitized? this makes no sense at all

Reply:
folks - the goal of logging is to log what the user is inputting. it's in fact more risky to give a user input to a regex than it is to log it. please when making such changes explain exactly what the issue is.

Reply:
https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS